src/content/blog/astro-db-deep-dive.mdx

```diff
@@ -0,0 +1,77 @@
+---
+title: "Astro DB: A Deep Dive"
+description: "Yesterday we launched a fully-managed SQL database service designed exclusively for the Astro web framework. Let's dive into the implementation details of Astro DB: how it works, why we built it, and why we're adopting libSQL."
+publishDate: "March 13, 2024"
+authors:
+  - matthew
+coverImage: "/src/content/blog/_images/astro-db-deep-dive/header-astro-db-deep-dive.webp"
+socialImage: "/src/content/blog/_images/astro-db-deep-dive/og-astro-db-deep-dive.webp"
+lang: "en"
+---
+
+**Yesterday we launched a fully-managed SQL database service designed exclusively for the Astro web framework. Let's dive into the implementation details of [Astro DB](https://astro.build/db/): how it works, why we built it, and why we're adopting libSQL.**
+
+## How we got here
+
+Astro is unique for its focus on building **content-driven websites.** The center of this is, of course, content, which is why in Astro 2.0 we shipped [Content Collections](https://astro.build/blog/introducing-content-collections/). Our users loved it as a way to manage their local content.
+
+WordPress has always been a huge inspiration for us. One of the things that makes WordPress so special is its **built-in** **database**. You're not just managing your article content, you're managing data, pages, blocks, images, and an entire ecosystem of plugins.
+
+We wanted something like this for Astro, but we quickly realized that we'd reached the limits of how far static repo data could take us.
+
+## Finding (and losing) SQLite
+
+In an effort to evolve Content Collections with data collections and references, we realized what we were doing was building a database-like ORM from the filesystem, and we ran into several challenges doing so. Core team member Erika Moreau brought up the idea: why don't we just use a database, [SQLite](https://www.sqlite.org/)?
+
+We fell in love with the idea of a lightweight database built into Astro itself. SQLite was perfect for the read-heavy workloads of most content-driven websites. 
+
+We prototyped the idea but ran into a few blockers. SQLite is a C library, so it needs native add-ons to run in Node.js. This is ok for local development, but native add-ons are difficult to deploy to serverless hosts and the startup time was worrisome. Additionally key environments like [StackBlitz](https://stackblitz.com/) would fail to run it entirely.
```

Comment:
comma nit
```suggestion
We prototyped the idea but ran into a few blockers. SQLite is a C library, so it needs native add-ons to run in Node.js. This is ok for local development, but native add-ons are difficult to deploy to serverless hosts and the startup time was worrisome. Additionally, key environments like [StackBlitz](https://stackblitz.com/) would fail to run it entirely.
```